### PR TITLE
owner, util: use `timeutil.Sleep()` instead of `sleepContext`

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: not needed

Problem Summary: ref https://www.youtube.com/watch?v=qq3gu0JQ0yU&t=20m17s, use a new function `timeutil.Sleep()` instead of `sleepContext`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: util/timeutil/time.go [hunk 1]
```diff
@@ -259,3 +260,17 @@ func ParseTimeZone(s string) (*time.Location, error) {
 
 	return nil, ErrUnknownTimeZone.GenWithStackByArgs(s)
 }
+
+// Sleep is a function like time.Sleep(),
+// the difference is it will return early when ctx finished.
+func Sleep(ctx context.Context, d time.Duration) error {
```
Comment (bb7133) [inline util/timeutil/time.go:266]: For now all funcs in `time.go` are related to timezones, so it's a bit weird to add `Sleep` here, how about:

* Either we could rename `time.go` to something like `tzutil.go` and add `Sleep` to a new `time.go`.
* Or we could add `Sleep` to `util/misc.go`

WDYT?

Comment (Defined2014) [reply util/timeutil/time.go:24]: I'm prefer the first one and think maybe named `time_zone.go` is better.

Updated.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**